service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceAuthenticationWithSharedAccessPolicyKey.java

```diff
@@ -11,15 +11,15 @@
  * ServiceAuthenticationWithSharedAccessPolicyKey type policy on
  * given IotHubConnectionString object.
  */
-public class ServiceAuthenticationWithSharedAccessPolicyKey extends AuthenticationMethod
+class ServiceAuthenticationWithSharedAccessPolicyKey extends AuthenticationMethod
 {
     /**
      * Constructor to create instance from policy name and policy key
      *
      * @param policyName The policy name string
      * @param key The policy key string
      */
-    public ServiceAuthenticationWithSharedAccessPolicyKey(String policyName, String key)
+    ServiceAuthenticationWithSharedAccessPolicyKey(String policyName, String key)
```

Comment:
private?

Reply:
This constructor is called from another class within this package, so I can't mark it as private. This lack of modifier makes the constructor "Package private" which means only classes within this package can access it. Classes outside of the package, even if they extend this class, can't access it